flow/scripts/resize.tcl

```diff
@@ -27,9 +27,11 @@ puts ""
 set_dont_use $::env(DONT_USE_CELLS)
 
 # Do not buffer chip-level designs
-if {![info exists ::env(FOOTPRINT)]} {
-  puts "Perform port buffering..."
-  buffer_ports
+if {![info exists ::env(FOOTPRINT)] } {
+  if { ![info exists ::env(PORTS_BUFFER)] || $::env(PORTS_BUFFER) == 1 } {
```

Comment:
Please add `PORTS_BUFFER` to part of [Flow_Variables.md](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/docs/user/FlowVariables.md) with valid description.

Reply:
Is there any more fine grained control available?

Will input/output buffers be inserted manually?

We talked about how input/output buffering could be removed if the boundry conditions (in .sdc?) were carefully described.

Reply:
Would prefer the name `BUFFER_PORTS` over `PORTS_BUFFER`

Reply:
Maybe the name should be DONT_BUFFER_PORTS?

Reply:
Negative in config variables is not great...

How about adding a default value to .mk files and let the user override? Simplified .tcl too.

Reply:
I think I agree with @tspyrou here, it's easier and safer to check for the existence of a negative rather than set a default across all relevant scripts.

Reply:
I prefer a positive as well.  Why not just set the default in the Makefile where it is global?

Reply:
> I think I agree with @tspyrou here, it's easier and safer to check for the existence of a negative rather than set a default across all relevant scripts.

It is trivial to set a default across all scripts, just set it in Makefile or perhaps platform config.mk.  There are lots of examples already.

Reply:
Example:

```

Reply:
Changed variable to DONT_BUFFER_PORT.

The resize.tcl script was not written to take into consideration of enabling / disabling the feature. It query other variable ....

So ... just to make the script work with all platforms as is, we are introducing this new variable.

I agreed with default variable setting. Will add in the default DONT_BUFFER_PORT = 0.

The PR is ready for another review.

Reply:
I think that the need for this variable should go away when we improve the automatic code. In the meantime the negative seems ok to me but I will defer to the wisdom of the group.

Reply:
How far out is an automatic solution that make this option moot?

Reply:
@tspyrou What automatic solution are we discussing?

Reply:
I think the point of having the negative `DONT_BUFFER_PORTS` was so that there didn't need to be a default. The script could check for existence of the envar only. If there's going to be a default variable, might as well make it a positive default on `BUFFER_PORTS=1` rather than a default negative off `DONT_BUFFER_PORTS=0`.

Reply:
I think this has reach the level of bike shedding and we should just merge and move on.

Reply:
Agreed, though: I'm inclined to say that there are concerns separate from those that @louiic is trying to address that can be addressed individually in future pull requests....

Reply:
> bike shedding

Learned a new term today 🙂